.github/workflows/helm-lint-test.yaml

```
@@ -0,0 +1,161 @@
+    outputs:
+      changed: "${{ steps.list-changed.outputs.changed }}"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Helm
+      uses: azure/setup-helm@v3
+      with:
+        version: ${{ github.event.inputs.helm_version || 'latest' }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+        check-latest: true
+
+    - name: Set up chart-testing
+      uses: helm/chart-testing-action@v2.6.1
+
+    - name: Run chart-testing (list-changed)
+      id: list-changed
+      run: |
+        changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch }})
+        if [[ -n "$changed" ]]; then
+          echo "changed=true" >> $GITHUB_OUTPUT
+        fi
+
+  lint-test:
+    runs-on: ubuntu-latest
+    needs: [list-changed]
+    if: ${{ github.event_name != 'pull_request' || needs.list-changed.outputs.changed == 'true' }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Helm
+      uses: azure/setup-helm@v3
+      with:
+        version: ${{ github.event.inputs.helm_version || 'latest' }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+        check-latest: true
+
+    - name: Kubernetes KinD Cluster
+      uses: container-tools/kind-action@v2
+      with:
+        # upgrade version, default (v0.17.0) uses node image v1.21.1 and doesn't work with more recent node image versions
+        version: v0.20.0
+        # default value for event_name != workflow_dispatch
+        node_image: ${{ github.event.inputs.node_image || 'kindest/node:v1.27.3' }}
+
+    - name: Set up chart-testing
+      uses: helm/chart-testing-action@v2.6.1
+
+    - name: Lint chart
+      run: ct lint --validate-maintainers=false --target-branch ${{ github.event.repository.default_branch }}
+    
+    - name: Download chart dependencies (recursive)
+      shell: bash
+      run: hack/helm-dependencies.bash
+
+    - name: Install chart and run tests
+      run: ct install --charts charts/umbrella
+
+    ## Skip upgrade for now until a working chart is released
+    #- name: Run helm upgrade
+    #  run: |
+    #    helm repo add bitnami https://charts.bitnami.com/bitnami
+    #    helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
+    #    helm install [NAME] tractusx-dev/[CHART] --version ${{ github.event.inputs.upgrade_from || 'x.x.x' }}
+    #    helm dependency update charts/[CHART]
+    #    helm upgrade [NAME] charts/[CHART]
+
+  kyverno:
+    runs-on: ubuntu-latest
+    needs: [list-changed]
+    if: ${{ github.event_name != 'pull_request' || needs.list-changed.outputs.changed == 'true' }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
```

Why do we need all commits for Kyverno?

```suggestion
    - name: Checkout
      uses: actions/checkout@v4
```